mayastor/src/bdev/dev/nvmx/controller.rs

```diff
@@ -328,6 +330,70 @@ impl<'a> NvmeController<'a> {
         debug!("{} all namespaces removed", self.name);
     }
 
+    /// Get I/O statistics for all I/O channels of the controller.
+    pub fn get_io_stats<T: 'static + Sized, F>(
+        &self,
+        cb: F,
+        cb_arg: T,
+    ) -> Result<(), CoreError>
+    where
+        F: Fn(Result<BlockDeviceIoStats, CoreError>, T) + 'static,
```

Comment:
just an idea. By pushing the oneshot channel from io_stats() to this routine, you could keep larger part of the rust code clean (async method without without callbacks). But maybe it cannot be done for some reason.

Reply:
Unfortunately, we can't use this method as async. We invoke controller's methods under protection of the mutex, and await()'ing under mutex is not always a good idea. In fact, our linter complains about such a thing. That's why we initiate an async operation and then wait with the controller freed.